resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/repository/PersistenceLayer.java

```diff
@@ -33,61 +33,58 @@
      * Given a resource id, return all {@link LeafResource}s underneath it
      *
      * @param resourceId the resource id to query
-     * @return Optional.of a {@link Stream} of {@link LeafResource}s if the persistence store is aware of these resources
-     * Optional.empty if no such information has been persisted
-     * nb. the {@link Stream} may still be empty if this resourceId is an empty directory
+     * @return {@link Iterator} of {@link LeafResource}s if the persistence store is aware of these resources
      */
-    Optional<Stream<LeafResource>> getResourcesById(String resourceId);
+    Iterator<LeafResource> getResourcesById(String resourceId);
```

Comment:
It used to return Optional.of(Stream) / Optional.empty(), now it just returns an Iterator
What does it return in the case where it used to return the Optional.empty()?

Reply:
Updated to return an `Optional`